New addon: MountMuzzle

Sjshovan opened the pull request:
Allows players to change or remove the default mount music.

Sjshovan committed:
Add MountMuzzle addon

Sjshovan committed:
Fix: Zone music gets silenced if player enters reive on mount with zone muzzle selected.
Fix: Player reaches error if no arguments are given upon invoking the addon.
Update: Convert tab characters to spaces, simplify code.
Update: README Usage instructions updated.
Update: README Known Issues updated.
Add: Table of Contents added to README.
Add: Prerequisites added to README.
Add: Installation added to README.
Add: Support added to README.
Add: License added to README.

z16 commented on addons/MountMuzzle/data/settings.xml:
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.1" ?>
```
Please remove this file, it will be auto-generated. You should never check anything into the `data` sub-folder of an addon, as that is an entirely client-side managed folder.

Sjshovan replied:
Done.

z16 commented on addons/MountMuzzle/LICENSE:
```diff
@@ -0,0 +1,25 @@
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL Patrick Finnigan BE LIABLE FOR ANY
```
Not sure if that is you, but the person or organization in this place should match the copyright holder at the top of the notice.

Sjshovan replied:
Done.

z16 commented on addons/MountMuzzle/README.md:
```diff
@@ -0,0 +1,155 @@
+    //lua load mountmuzzle  
+    //lua l mountmuzzle
+    
+**Automatically load** this addon by adding `lua load mountmuzzle` or `lua l mountmuzzle` to the bottom of your `Windower4/scripts/init.txt` file.
```
This is actually discouraged. Automatic loading should be handled either through the Windower launcher or through the `plugin_manager` addon, which offers finer grained control over character-based plugin and addon usage.

Sjshovan replied:
Done.

z16 commented on addons/MountMuzzle/README.md:
```diff
@@ -0,0 +1,155 @@
+___
+### Support
+**Having Issues with this addon?**
+* Please let me know [here.](https://github.com/Ap0gee/MountMuzzle/issues/new)
```
It bothers me more than it should that the period is part of the link :<

Sjshovan replied:
Done.

z16 commented on addons/MountMuzzle/constants.lua:
```diff
@@ -0,0 +1,94 @@
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL Patrick Finnigan BE LIABLE FOR ANY
```
See above.

Sjshovan replied:
Done.

z16 commented on addons/MountMuzzle/helpers.lua:
```diff
@@ -0,0 +1,80 @@
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL Patrick Finnigan BE LIABLE FOR ANY
```
See above.

z16 commented on addons/MountMuzzle/mountmuzzle.lua:
```diff
@@ -0,0 +1,222 @@
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL Patrick Finnigan BE LIABLE FOR ANY
```
See above.

Sjshovan replied:
Done.

z16 commented on addons/MountMuzzle/helpers.lua:
```diff
@@ -0,0 +1,80 @@
+
+function ucFirst(str)
+    return (str:gsub("^%l", string.upper))
+end
```
There's already ` str:ucfirst()` function in our `strings` library.

Sjshovan replied:
Done.

z16 commented on addons/MountMuzzle/helpers.lua:
```diff
@@ -0,0 +1,80 @@
+    windower.add_to_chat(color, response)
+    windower.console.write(response:strip_colors())
+end
+
```
While it's not terribly important, nor am I not gonna merge this because of it, your modular file design can be improved. Currently you use some functions in here that you don't declare in here. Thus you rely on this file being `require`d after some others. Normally you'd `require` everything you use in this file directly in this file.

And the `constants` file would not just write global values, but return them:
```lua
return packets, player, music, colors, muzzle
```

And in this file you'd set them like this:
```lua
local packets, player, music, colors, muzzle = require('constants')`
```

Again, don't need to really bother with it, but that's what a really modular design would look like, as now you rely on the order of inclusion too much.

Sjshovan replied:
Instead of returning everything like: `return function() return packets, player, music, colors, muzzles end` I return a table so I can get specific objects from the file like: `local colors = require(constants').colors` in `helpers.lua` and still require everything within main.  Is the approach ok?

z16 replied:
Yes, that's fine, and like I said it wasn't *bad* before either, was just a suggestion to keep the logic more separated and modular. This is totally fine.

z16 commented on addons/MountMuzzle/mountmuzzle.lua:
```diff
@@ -0,0 +1,222 @@
+
+windower.register_event('addon command', function(command, ...)
+    if command then
+        local command = command:lower()
```
You probably want to remove the `local`. Otherwise this statement does nothing. It creates a local variable `command` which *hides* the original `command` variable, and which is immediately discarded, as the inner `command` dies at the `else` keyword.

Sjshovan replied:
Done.

z16 commented on addons/MountMuzzle/mountmuzzle.lua:
```diff
@@ -0,0 +1,222 @@
+end
+
+function injectMuzzleMusic()
+    windower.packets.inject_incoming(
```
You can use the packets library to do the packing and conversions for you:
```lua
    packets.inject(packets.new('incoming', packets.inbound.music_change.id, {
        ['BGM Type'] = music.types.mount,
        ['Song ID'] = resolveCurrentMuzzle().song,
    }))
```

Sjshovan replied:
Yes! Have been using this for next mod, thank you! Done.

z16 reviewed (changes requested):
Most of it looks good, but please look through the copyright comments I made. If that is correct then this is ready to merge (the other things I wrote were just suggestions). Let me know when you're ready to have this merged.

Sjshovan committed:
Remove: Removed /data/settings.xml file.
Update: licences now display correct author name.
Update: README refactored  in attempt to meet merge criteria.
Update: helpers.lua now requires only colors from constants.lua.
Update: constants.lua now returns table of globals for modular support.
Update: mountmuzzle.lua refactored in attempt to meet merge criteria.
